CMakeLists.txt

```diff
@@ -56,6 +64,11 @@ add_subdirectory(api)
 add_subdirectory(src)
 
 if(ICEBERG_BUILD_TESTS)
+  fetchcontent_declare(googletest
```

Comment:
This adds a GoogleTest dependency. Was there a decision to use GT? A popular alternative is https://github.com/catchorg/Catch2/, I'm not sure which one would be better.

Reply:
Yeah, I see that Catch2 has TDD and BDD, which is better, I will do some work and find out which one fits us better.

Reply:
Perhaps open a GH issue to discuss it?